tests/unit_tests/libs/helpers/locking.py

```diff
@@ -0,0 +1,22 @@
+import pytest
+from mindsdb_native.libs.helpers.locking import MDBLock
+import portalocker
+
+
+class ExceptionForTest(BaseException): pass
+
+
+def func():
+    raise ExceptionForTest('random exception')
+
+
+def test_locking():
+    for flags in ['shared', 'exclusive']:
+        lock = MDBLock(flags, 'name')
+
+        with pytest.raises(ExceptionForTest):
+            lock(func)()
+
+        with pytest.raises(ExceptionForTest):
+            with lock:
+                func()
```

Comment:
I don't understand: does this assert something? Does this test validate that the lock is released properly?

Reply:
I *assume* it asserts (in the `exclusive` case) that the lock has been properly released previously.

But I agree that this test doesn't seem to do anything in the `shared` case.

@maximlopin maybe add some comments as to what the test is doing, or just remove the `shared` case and we can merge the PR without it.

*Also, might be cleaner to define `func` within `test_locking`.